### PR TITLE
Add explicit_subtransactions in plcontainer.

### PR DESCRIPTION
**Pull request by zhangh43**
This is previous work to support subtransaction in plcontainer python(consistent with plpython upstream).

### CODE REVIEW

File: src/plcontainer.c [hunk 1]
```diff
@@ -172,81 +191,99 @@ static plcProcResult *plcontainer_get_result(FunctionCallInfo  fcinfo,
...
-
-		if (res < 0) {
-			elog(ERROR, "Error sending data to the client: %d. "
-				"Maybe retry later.", res);
-			return NULL;
+    int volatile save_subxact_level = list_length(explicit_subtransactions);
+
+    	PG_TRY();
+    	{
+		req = plcontainer_create_call(fcinfo, pinfo);
```
Comment (HaozhouWang) [inline src/plcontainer.c]: indent

File: src/plcontainer.c [hunk 2]
```diff
@@ -360,3 +397,42 @@ static void plcontainer_process_exception(plcMsgError *msg) {
...
+		CurrentResourceOwner = subtransactiondata->oldowner;
+		plcontainer_free(subtransactiondata);
+	}
+}
+
+/* define this away */
+static void
+plcontainer_free(void *ptr)
+{
+	pfree(ptr);
```
Comment (HaozhouWang) [inline src/plcontainer.c]: Do we need to check whether ptr is NULL?

Comment (zhangh43) [reply src/plcontainer.c]: @paul-guo-  consistent with upstream plpython PLy_free

**Comment by zhangh43** [comment]
Close this PR, merge the work in #PR136

**Pull request closed without merging**